# Electronegativity by element group

**Instructor solution.** Run this notebook independently from top to bottom.

In [1]:
from pathlib import Path
import sys

# Support kernels started in the repository root or in instructor-notebooks.
ROOT = Path.cwd()
if not (ROOT / "data").is_dir():
    ROOT = ROOT.parent
if not (ROOT / "data").is_dir():
    raise FileNotFoundError("Open this notebook from the repository or instructor-notebooks folder.")
sys.path.insert(0, str(ROOT))

In [2]:
import pandas as pd
df = pd.read_csv(ROOT / "data/PubChemElements_all.csv")
groups = df.groupby("GroupBlock")
groups.describe()

AtomicNumber                                           \
                             count       mean        std   min    25%   50%   
GroupBlock                                                                    
Actinide                      15.0  96.000000   4.472136  89.0  92.50  96.0   
Alkali metal                   6.0  35.333333  31.506613   3.0  13.00  28.0   
Alkaline earth metal           6.0  36.333333  31.506613   4.0  14.00  29.0   
Halogen                        6.0  52.666667  41.654131   9.0  21.50  44.0   
Lanthanide                    15.0  64.000000   4.472136  57.0  60.50  64.0   
Metalloid                      7.0  38.714286  26.468310   5.0  23.00  33.0   
Noble gas                      7.0  46.285714  42.746206   2.0  14.00  36.0   
Nonmetal                       7.0  12.428571  10.845232   1.0   6.50   8.0   
Post-transition metal         11.0  77.000000  36.649693  13.0  49.50  82.0   
Transition metal              38.0  61.736842  31.977766  21.0  32.25  47.5   

                                     AtomicMass              ... BoilingPoint  \
                          75%    max      count        mean  ...          75%   
GroupBlock                                                   ...                
Actinide                99.50  103.0       15.0  245.734139  ...     4289.500   
Alkali metal            50.50   87.0        6.0   85.080159  ...     1156.000   
Alkaline earth metal    51.50   88.0        6.0   87.392099  ...     2066.750   
Halogen                 77.00  117.0        6.0  127.575176  ...      363.350   
Lanthanide              67.50   71.0       15.0  155.946857  ...     3610.500   
Metalloid               51.50   84.0        7.0   92.112717  ...     3322.000   
Noble gas               70.00  118.0        7.0  113.629169  ...      153.755   
Nonmetal                15.50   34.0        7.0   26.434109  ...      837.875   
Post-transition metal  113.50  116.0       11.0  191.760903  ...     2634.500   
Transition metal        79.75  112.0       38.0  151.042093  ...     4701.000   

                               Density                                  \
                           max   count       mean       std        min   
GroupBlock                                                               
Actinide               5061.00     9.0  15.266667  3.641548  10.070000   
Alkali metal           1615.00     5.0   1.170800  0.554548   0.534000   
Alkaline earth metal   2744.00     6.0   2.731667  1.350828   1.540000   
Halogen                 457.55     5.0   3.008982  3.070307   0.001696   
Lanthanide             3793.00    15.0   7.688667  1.277788   5.240000   
Metalloid              4273.00     7.0   5.433657  2.465004   2.329600   
Noble gas               211.45     6.0   0.003702  0.003607   0.000179   
Nonmetal               4098.00     7.0   1.566538  1.764103   0.000090   
Post-transition metal  2875.00     7.0   8.022286  3.222662   2.700000   
Transition metal       5869.00    29.0  11.557572  5.719426   2.990000   

                                                                  
                             25%        50%        75%       max  
GroupBlock                                                        
Actinide               13.510000  14.000000  18.950000  20.25000  
Alkali metal            0.890000   0.970000   1.530000   1.93000  
Alkaline earth metal    1.767500   2.245000   3.375000   5.00000  
Halogen                 0.003214   3.110000   4.930000   7.00000  
Lanthanide              6.835000   7.520000   8.675000   9.84000  
Metalloid               3.846500   5.776000   6.458500   9.32000  
Noble gas               0.001121   0.002758   0.005348   0.00973  
Nonmetal                0.001340   1.820000   2.167000   4.80900  
Post-transition metal   6.598500   7.310000  10.574500  11.80000  
Transition metal        7.300000  10.200000  13.533600  22.57000  

[10 rows x 72 columns]

In [3]:
electronegativity_stats = groups["Electronegativity"].describe()
electronegativity_stats

,count,mean,std,min,25%,50%,75%,max
GroupBlock,,,,,,,,
Actinide,15.0,1.308000,0.079929,1.10,1.3000,1.300,1.3000,1.50
Alkali metal,6.0,0.840000,0.100598,0.70,0.7975,0.820,0.9025,0.98
Alkaline earth metal,6.0,1.103333,0.276237,0.89,0.9125,0.975,1.2325,1.57
Halogen,5.0,2.992000,0.660091,2.20,2.6600,2.960,3.1600,3.98
Lanthanide,11.0,1.188182,0.058791,1.10,1.1350,1.200,1.2350,1.27
Metalloid,7.0,2.040000,0.086987,1.90,2.0050,2.040,2.0750,2.18
Noble gas,2.0,2.800000,0.282843,2.60,2.7000,2.800,2.9000,3.00
Nonmetal,7.0,2.650000,0.450037,2.19,2.3750,2.550,2.8100,3.44
Post-transition metal,7.0,1.875714,0.252907,1.61,1.7000,1.810,1.9900,2.33


Halogens have the highest group mean (2.992); alkali metals have the lowest (0.840). Nonmetals generally have higher electronegativity than metals, consistent with their greater attraction for bonding electrons. Group means hide variation within each category. Missing values are excluded: the noble-gas mean of 2.8 uses only two values, so it should not be generalized to every noble gas.